pkg/cfn/builder/nodegroup.go

```diff
@@ -459,6 +459,9 @@ func newLaunchTemplateData(ctx context.Context, n *NodeGroupResourceSet) (*gfnec
 		UserData:          gfnt.NewString(userData),
 		MetadataOptions:   makeMetadataOptions(ng.NodeGroupBase),
 		TagSpecifications: makeTags(ng.NodeGroupBase, n.options.ClusterConfig.Metadata),
+		EnclaveOptions: &gfnec2.LaunchTemplate_EnclaveOptions{
+			Enabled: gfnt.NewBoolean(ng.EnclaveEnabled),
+		},
```

Comment:
We should only add this `EnclaveOptions` part if the EnclaveEnabled boolean was explicitly provided. Otherwise, this will default to always including this block in the cloudformation which could have implications for unsupported regions, etc.